### PR TITLE
feat(skrapeit/skrape.it): add usage examples

### LINKED ISSUE #42: Sample wanted for skrapeit/skrape.it
## About the library

- *Description*:  A Kotlin-based HTML / XML deserialization library that places particular emphasis on ease of use and a high level of readability by providing an intuitive DSL.
- *GitHub*: https://github.com/skrapeit/skrape.it
- *Official website*: https://docs.skrape.it/docs/

## What the sample could do

Parse an HTML file and verify that we can find a given information in it

## Wants to contribute the sample?

- Say so in a comment here
- Follow instructions in [CONTRIBUTING.md](https://github.com/LouisCAD/kotlin-libraries-playground/blob/main/CONTRIBUTING.md)
- Submit your pull request!

### PR DESCRIPTION
**Pull request by virtuoushub**
Parses an HTML, string, file, and response to verify that we can find given information

closes #42

### CODE REVIEW

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    println("Parse and verify local HTML resource")
+    `Parse and verify local HTML resource`()
+    println("Parse and verify response of url")
+    `Parse and verify response of url`()
+}
+
+/**
+ * [Picking Html-Elements from a Doc](https://docs.skrape.it/docs/dsl/parsing-html#picking-html-elements-from-a-doc)
+ */
+private fun `Pick Html-Elements from a Doc`() {
```
Comment (jmfayard) [inline kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt:43]: Can you split this example in two parts?
Both the example and the code that decode it takes so much vertical space that it's hard to see them at the same time to understand what is being done.
Apart from that, great job 👏🏻

**Review by jmfayard — APPROVED** [review]

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt (hunk 1 above)
Comment (virtuoushub) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt:43]: I have updated the code. Does it match what you were requesting?

Comment (jmfayard) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt:43]: Ah not really, I'm sorry that I was unclear. 

I wanted to split in two parts

- a part `<html><head>....</head></html>` with the associated Kotlin code
- a part `<html><body>...</body></html>` with the associated Kotlin code

This way we can look at both the HTML and the Kotlin code on the same screen without having to scroll back and forth

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+private fun loadReferenceResult(): String {
+    return ForTestingLocalResourceLoading.javaClass
+        .classLoader
+        .getResource("references/kotlinx.html/ref.html")!!
+        .readText()
+}
+
+private fun getMockHtml(): String {
+    return """
```
Comment (LouisCAD) [inline kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt:183]: Could use expression body syntax, and use `trimdIndent()` as well?

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt (hunk 1 above)
Comment (virtuoushub) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt]: Understood, that makes more sense. 

I have split the examples out a bit. However, in the interest of space (and maybe leaving it up as an exercise for the reader) I did not fully show what getting all the different elements/attributes in the HTML's `<body>` look like.

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt [hunk 3]
```diff
@@ -179,8 +179,8 @@ private fun loadReferenceResult(): String {
         .readText()
 }
 
-private fun getMockHtml(): String {
-    return """
+private fun getMockHtml(): String =
+    """
```
Comment (LouisCAD) [inline kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt]: Why keep the line break here?

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt (hunk 1 above)
Comment (jmfayard) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt]: That's fine, the samples are not intended as a complete reference,
the goal is to get people interested in the library,
and give them a starting point if they want to try out deeper themselves

**Comment by jmfayard** [comment]
LGTM!

File: kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt (hunk 3 above)
Comment (virtuoushub) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt]: I was not quite sure on the expected styling. Was doing my best to follow the code convention

> If the function has an expression body that doesn't fit in the same line as the declaration, put the = sign on the first line. Indent the expression body by four spaces.

https://kotlinlang.org/docs/reference/coding-conventions.html#expression-body-formatting

Comment (LouisCAD) [reply kotlin-jvm/src/main/kotlin/playground/SkrapeIt.kt]: This is when the first line of the body doesn't fit on the line, but it's not the case here.
I submitted an edit to improve it the convention wording and example: https://github.com/JetBrains/kotlin-web-site/pull/1925

Can you open a new PR to improve this here?

**Pull request merged**